add new override examples (#4154)
aws-amplify/docs: src/pages/cli/project/override.mdx

|Amplify-generated resource|Description|
|-|-|
|[authRole](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html)|The IAM role for authenticated access to your app backend|
|[unauthRole](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html)|The IAM role for authenticated or guest access to your app backend|

> ## Customize unauthRole policies

Reviewer: ```suggestion
## Example: Modify unauthRole's IAM policies
```

> ## Customize UnAuth Role policies

Reviewer 2: Should this be  - unauthRole?

> For example, to use the project override to update the unauthRole policy for S3, use the following:

Reviewer: Can we pick another example. Other than S3? Especially because customers can configure this via `amplify update storage` without an override.

Author: updated with geo example

> For example if we want to use project override to update unAuth policy for S3, we can use the following example:

Reviewer 2: ```suggestion
For example, to use the project override to update the unAuth policy for S3, use the following:
```

Reviewer 2: see note above re: naming unauthRole?

> For example, to use the project override to update the unauthRole policy for Geo category, use the following:

Reviewer: ```suggestion
For example, use `amplify override project` to further modify the unauthRole policy for Geo category beyond the default policy statements:
```

```ts
import { AmplifyRootStackTemplate } from "@aws-amplify/cli-extensibility-helper";

export function override(resources: AmplifyRootStackTemplate) {
  const unauthRole = resources.unauthRole;

>   const basePolicies = Array.isArray(unauthRole.policies)

Reviewer: Can you elaborate more on what's happening here?

    ? unauthRole.policies
    : [unauthRole.policies];

  unauthRole.policies = [
    ...basePolicies,
    {
      policyName: "amplify-permissions-custom-resources",
      policyDocument: {
        Version: "2012-10-17",
        Statement: [
          {
            Resource: "<ARN of resource>",
            Action: ["s3:PutObject", "s3:GetObject", "s3:DeleteObject"],
            Effect: "Allow",
          },
        ],
      },
    },

>   ];

Reviewer: nit: given other examples don't have semi-colons, we should probably skip it here.
